src/main/java/org/zalando/nakadi/repository/db/TimelineDbRepository.java

```diff
@@ -0,0 +1,128 @@
+    public TimelineDbRepository(final JdbcTemplate jdbcTemplate, final ObjectMapper jsonMapper) {
+        super(jdbcTemplate, jsonMapper);
+    }
+
+    public List<Timeline> listTimelines(final String eventType) {
+        return jdbcTemplate.query(
+                BASE_TIMELINE_QUERY + " WHERE t.et_name=? order by t.tl_order",
+                timelineRowMapper,
+                eventType);
+    }
+
+    public Optional<Timeline> getTimeline(final UUID id) {
+        final List<Timeline> timelines = jdbcTemplate.query(
+                BASE_TIMELINE_QUERY + " WHERE t.tl_id=?", timelineRowMapper, id);
+        return Optional.ofNullable(timelines.isEmpty() ? null : timelines.get(0));
+    }
+
+    public Timeline createTimeline(final Timeline timeline) {
+        try {
+            jdbcTemplate.update(
+                    "INSERT INTO zn_data.timeline(" +
+                            " tl_id, et_name, tl_order, st_id, tl_topic, " +
+                            " tl_created_at, tl_switched_at, tl_cleanup_at, tl_latest_position) " +
+                            " VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?::jsonb)",
+                    timeline.getId(),
+                    timeline.getEventType(),
+                    timeline.getOrder(),
+                    timeline.getStorage().getId(),
+                    timeline.getTopic(),
+                    timeline.getCreatedAt(),
+                    timeline.getSwitchedAt(),
+                    timeline.getCleanupAt(),
+                    timeline.getLatestPosition() == null ? null :
+                            jsonMapper.writer().writeValueAsString(timeline.getLatestPosition()));
+            return timeline;
+        } catch (final JsonProcessingException ex) {
+            throw new IllegalArgumentException("Can not serialize latest position to json", ex);
+        }
+    }
+
+    public void updateTimelime(final Timeline timeline) {
+        try {
+            jdbcTemplate.update(
+                    "UPDATE zn_data.timeline SET " +
+                            " et_name=?, " +
+                            " tl_order=?, " +
+                            " st_id=?, " +
+                            " tl_topic=?, " +
+                            " tl_created_at=?, " +
+                            " tl_switched_at=?, " +
+                            " tl_cleanup_at=?, " +
+                            " tl_latest_position=?::jsonb " +
+                            " WHERE tl_id=?",
+                    timeline.getEventType(),
+                    timeline.getOrder(),
+                    timeline.getStorage().getId(),
+                    timeline.getTopic(),
+                    timeline.getCreatedAt(),
+                    timeline.getSwitchedAt(),
+                    timeline.getCleanupAt(),
+                    timeline.getLatestPosition() == null ? null :
+                            jsonMapper.writer().writeValueAsString(timeline.getLatestPosition()),
+                    timeline.getId());
```

Comment:
@v-stepanov Exception will be thrown. I use `@Nullable` here to support fake timelines, which do not have timeline id (will be introduced later). And of course it is incorrect to use this kind of timeline for database updates.